tests/test_sparse_tensors.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+
+# needed to register custom ops
+import xformers  # noqa: F401
+from xformers.ops import masked_matmul
+from xformers.sparse import BlockSparseTensor
+
+cuda_only = pytest.mark.skipif(not torch.cuda.is_available(), reason="requires CUDA")
+_devices = ["cuda:0"] if torch.cuda.is_available() else []
+
+
+def _create_tensor(device, BLOCK=32, Z=8, C=2, H=512, W=512, dtype=torch.float32):
+    layout = torch.randint(2, (C, H // BLOCK, W // BLOCK))
+    values = torch.randn(Z, layout.sum(), BLOCK, BLOCK, device=device).to(dtype)
+
+    mask = (
+        layout[None, :, :, None, :, None]
+        .repeat(Z, 1, 1, BLOCK, 1, BLOCK)
+        .reshape(Z, C, H, W)
+    )
+
+    return BlockSparseTensor(values, layout), mask.bool()
+
+
+@pytest.mark.parametrize("device", _devices)
+def test_masked_matmul(device):
+    BLOCK = 32
+    N, C, H, W, L = 8, 2, 512, 512, 64
+    mask_block, _ = _create_tensor(device, BLOCK, N, C, H, W, dtype=torch.bool)
+    mask = mask_block.to_dense()
+
+    a = torch.randn(N, C, H, L, device=device)
+    b = torch.randn(N, C, W, L, device=device)
+
+    aa = a.clone()
+    bb = b.clone()
+
+    b = b.transpose(-2, -1)
+    bb = bb.transpose(-2, -1)
+
+    a.requires_grad_(True)
+    b.requires_grad_(True)
+    aa.requires_grad_(True)
+    bb.requires_grad_(True)
+
+    # res_gt = masked_matmul(a, b, mask)
+    res_gt = a @ b
+    # res_gt[~mask] = 0
+    res_gt = torch.where(mask, res_gt, torch.zeros_like(res_gt))
+    res = masked_matmul(aa, bb, mask_block)
+
+    res_dense = res.to_dense()
+    # res_dense[~mask] = float('-inf')
+
+    assert res.dtype == res_gt.dtype
+    assert torch.allclose(res_dense, res_gt)
+
+    res_gt.sum().backward()
+    res._blocksparse_values.sum().backward()
+    # TODO: this is not passing!!!
```

Comment:
is that only when a row is [0], or do you have other issues ?

Reply:
The failures here are due to https://github.com/openai/triton/issues/419

Are we planning on releasing a new version of `1.1.x` sometime soon?